lightly/api/api_workflow_client.py

```diff
@@ -118,7 +118,7 @@ def dataset_id(self) -> str:
         try:
             return self._dataset_id
         except AttributeError:
-            all_datasets: List[DatasetData] = self._datasets_api.get_datasets()
+            all_datasets: List[DatasetData] = self.get_datasets()
```

Comment:
I guess this should be `get_all_datasets` as well?

Reply:
It depends, this is a bit of a weird function anyways... but I chose to only return the owned datasets so as to not accidentally set the dataset id to a shared dataset. I'd leave it like this if that's ok with you.